drivers/flash/soc_flash_nrf_ticker.c

```diff
@@ -58,18 +58,20 @@ static void time_slot_callback_work(uint32_t ticks_at_expire,
 	struct flash_op_desc *op_desc;
 	uint8_t instance_index;
 	uint8_t ticker_id;
+	int rc;
 
 	__ASSERT(ll_radio_state_is_idle(),
 		 "Radio is on during flash operation.\n");
 
 	op_desc = context;
-	if (op_desc->handler(op_desc->context) == FLASH_OP_DONE) {
+	rc = op_desc->handler(op_desc->context);
+	if (rc != FLASH_OP_ONGOING) {
 		ll_timeslice_ticker_id_get(&instance_index, &ticker_id);
 
 		/* Stop the time slot ticker */
 		_ticker_stop(instance_index, 0, ticker_id);
 
-		_ticker_sync_context.result = 0;
+		_ticker_sync_context.result = (rc == FLASH_OP_DONE) ? 0 : rc;
```

Comment:
This should be one of constants from `errno.h` negated.

Reply:
@eugmes
OK, I would use `-ECANCELED`.
What about using -ECANCELED as the `handler()` return code instead of `FLASH_OP_ABORTED`?

Reply:
Yes, that's one I was thinking about.

Reply:
now  `-ECANCELED` is transferred from the handler error code